src/replication.c

```diff
@@ -1250,11 +1613,15 @@ empty_mgmt_q_of_replica(replica_t *r)
 {
 	mgmt_cmd_t *mgmt_cmd;
 	while ((mgmt_cmd = TAILQ_FIRST(&r->mgmt_cmd_queue))) {
+		mgmt_cmd->io_hdr->status = ZVOL_OP_STATUS_FAILED;
 		switch (mgmt_cmd->io_hdr->opcode) {
 			case ZVOL_OPCODE_SNAP_CREATE:
-				mgmt_cmd->io_hdr->status = ZVOL_OP_STATUS_FAILED;
 				handle_snap_create_resp(r, mgmt_cmd);
```

Comment:
not related to this PR..
As you found through code walk-through, lets add a comment that handle_snap_create_resp is picking header status from different header

Reply:
I have added it in my TODO list. Will share it with you.